### PR TITLE
Clean up RegionSelector, expose it as a UI DSL component

### PR DESCRIPTION
**Pull request by abrooksv**
## License
I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: jetbrains-core/src/software/aws/toolkits/jetbrains/ui/RegionSelector.kt [hunk 1]
```diff
@@ -31,7 +29,7 @@ class RegionSelector : ComboBox<AwsRegion>() {
     }
 
     fun setRegions(regions: List<AwsRegion>) {
-        comboBoxModel.items = regions
+        comboBoxModel.replaceAll(regions)
```
Comment (manodnyab) [inline jetbrains-core/src/software/aws/toolkits/jetbrains/ui/RegionSelector.kt:32]: Whats the behavior change after using replaceAll?

Comment (rli) [reply jetbrains-core/src/software/aws/toolkits/jetbrains/ui/RegionSelector.kt:32]: behavior should be the same, but we're no longer hacking in something that the `CollectionComboBoxModel` already implements

File: jetbrains-core/src/software/aws/toolkits/jetbrains/ui/RegionSelector.kt [hunk 2]
```diff
@@ -44,9 +42,21 @@ class RegionSelector : ComboBox<AwsRegion>() {
...
         }
 
-    private companion object {
-        val RENDERER = SimpleListCellRenderer.create<AwsRegion>("") {
+    companion object {
+        private val RENDERER = SimpleListCellRenderer.create<AwsRegion>("") {
             it.displayName
         }
+
+        fun Cell.regionSelector(prop: KMutableProperty0<AwsRegion>, serviceId: String? = null): CellBuilder<ComboBox<AwsRegion>> {
```
Comment (rli) [inline jetbrains-core/src/software/aws/toolkits/jetbrains/ui/RegionSelector.kt:50]: clarify that the service id filters the region list?

File: jetbrains-core/src/software/aws/toolkits/jetbrains/ui/RegionSelector.kt [hunk 3]
```diff
@@ -44,9 +42,21 @@ class RegionSelector : ComboBox<AwsRegion>() {
...
+            val regionProvider = AwsRegionProvider.getInstance()
+            val regions = when {
+                serviceId != null -> regionProvider.allRegionsForService(serviceId).values.toMutableList()
+                else -> regionProvider.allRegions().values.toMutableList()
+            }
+            val model = CollectionComboBoxModel(regions)
+            return comboBox(model, prop, RENDERER).applyToComponent {
```
Comment (rli) [inline jetbrains-core/src/software/aws/toolkits/jetbrains/ui/RegionSelector.kt:57]: is this preferred over reusing our own component?

**Review by rli — APPROVED** [review]

File: jetbrains-core/src/software/aws/toolkits/jetbrains/ui/RegionSelector.kt (hunk 1 above)
Comment (abrooksv) [reply jetbrains-core/src/software/aws/toolkits/jetbrains/ui/RegionSelector.kt:32]: Richard is correct

File: jetbrains-core/src/software/aws/toolkits/jetbrains/ui/RegionSelector.kt (hunk 3 above)
Comment (abrooksv) [reply jetbrains-core/src/software/aws/toolkits/jetbrains/ui/RegionSelector.kt:57]: I don't think we need to. long term we should move off .form files and so we shouldn't need our own

File: jetbrains-core/src/software/aws/toolkits/jetbrains/ui/RegionSelector.kt (hunk 2 above)
Comment (abrooksv) [reply jetbrains-core/src/software/aws/toolkits/jetbrains/ui/RegionSelector.kt:50]: Will added

**Review by rli — APPROVED** [review]

**Pull request merged**